process.cabal

```diff
@@ -44,9 +44,16 @@ library
     exposed-modules:
         System.Cmd
         System.Process
-    if impl(ghc)
-        exposed-modules:
-            System.Process.Internals
+        System.Process.Internals
+    other-modules: System.Process.Common
+    if os(windows)
+        other-modules: System.Process.Windows
+        build-depends: Win32 >=2.2 && < 2.4
+        extra-libraries: kernel32
+        cpp-options: -DWINDOWS
```

Comment:
Usage was inconsistent in the codebase previously around a few different ways of detecting Windows, and I've seen concerns raised in the past about the reliability of those macros. I feel more comfortable using the same logic for the Haskell code conditionals as the Cabal conditionals guiding the package dependencies and module selection.